File: app/src/main/java/edu/mills/overflo/MainActivity.java
package edu.mills.overflo;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.GridLayout;
import android.view.View;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {

    private static final int sideSize = 3;
    private static int size;
    private static TextView[] textViews;
    private static Grid g;

    public static void drawGrid() {
        for (final Square s : g.getSquares()) {
            int i = s.getId();
            textViews[i].setText(Integer.toString(s.getValue()));
        }
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        GridLayout gridLayout = findViewById(R.id.grid);
        size = sideSize * sideSize;
        gridLayout.setColumnCount(sideSize);
        textViews = new TextView[size];
        g = new Grid(sideSize);
        for (final Square s : g.getSquares()) {
            int i = s.getId();
            textViews[i] = new TextView(this);
            textViews[i].setText(Integer.toString(s.getValue()));
            textViews[i].setTextSize(100);
            textViews[i].setPadding(10, 10, 10, 10);
            textViews[i].setId(i);
            textViews[i].setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    g.increment(s);
                    drawGrid();
                }

                ;
            });
            gridLayout.addView(textViews[i]);
        }
    }
}

File: app/src/main/java/edu/mills/overflo/MainActivity.kt
package edu.mills.overflo

import android.graphics.Color
import android.os.Bundle
import android.view.View
import android.widget.GridLayout
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    private lateinit var textViews: Array<TextView?>
    private var size = 0
    private var g: Grid? = null
    private var next = 0

    private fun drawGrid() {
        for (s in g!!.squares) {
            val i = s.id
            textViews[i]!!.text = s.value.toString()
            if (s.player == 1) {
                textViews[i]!!.setTextColor(resources.getColor(R.color.blue, theme))
            }
            if (s.player == 2) {
                textViews[i]!!.setTextColor(resources.getColor(R.color.gold, theme))
            }
        }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val gridLayout = findViewById<GridLayout>(R.id.grid)
        size = sideSize * sideSize
        gridLayout.columnCount = sideSize
        textViews = arrayOfNulls(size)
        g = Grid(sideSize)
        for (s in g!!.squares) {
            val i = s.id
            textViews[i] = TextView(this)
            textViews[i]!!.text = s.value.toString()
            textViews[i]!!.textSize = 80f
            textViews[i]!!.setPadding(5, 5, 5, 5)
            textViews[i]!!.id = i
            textViews[i]!!.setTextColor(Color.GRAY)
            textViews[i]!!.setOnClickListener {
                next = g!!.oneTurn(s)
                val colorView = findViewById<View>(R.id.turn_color)
                val textView = findViewById<TextView>(R.id.turn_text)
                if (next == 1) {
                    colorView.setBackgroundColor(resources.getColor(R.color.blue))
                    textView.setText(R.string.player_1_turn)
                }
                if (next == 2) {
                    colorView.setBackgroundColor(resources.getColor(R.color.gold))
                    textView.setText(R.string.player_2_turn)
                }
                drawGrid()
            }
            gridLayout.addView(textViews[i])
        }
    }

    companion object {
        private const val sideSize = 4
    }
}
